### PR TITLE
Add X Macro for fun and for profit

### PR DESCRIPTION
**Pull request by zadjii-msft**
**Summary of the Pull Request**

This PR adds an X Macro for defining our ShortcutActions. This means that you can add the action in one place, and have the macro synthesize all sorts of boilerplate for you!

From the `AllShortcutActions.h` file:

> For a clearer explanation of how this file should be used, see:
> https://en.wikipedia.org/wiki/X_Macro
>
> Include this file to be able to quickly define some code in the exact same
> way for _every single shortcut action_. To use:
>
> 1. Include this file
> 2. Define the ON_ALL_ACTIONS macro with what you want each action to show up
>    as. Ex:
>
>    #define ON_ALL_ACTIONS(action) void action##Handler();
>
> 3. Then, use the ALL_SHORTCUT_ACTIONS macro to get the ON_ALL_ACTIONS marcro
>    repeated once for every ShortcutAction
>
> This is used in KeyMapping.idl, ShortcutAction.*, TerminalPage.*, etc. to
> reduce the number of places where we must copy-paste boiler-plate code for
> each action. This is _NOT_ something that should be used when any individual
> case should be customized.

**PR Checklist**
* [x] Scratches an itch
* [x] I work here
* [x] Tests passed
* [n/a] Requires documentation to be updated

**Detailed Description of the Pull Request / Additional comments**

Originally I had this blocked as a follow up to #9662. However, I've grown tired after a month of merging main into this branch, and I'm just shipping it separately. It will inevitably conflict with anyone who has actions in flight currently.

**Validation Steps Performed**
The code still builds exactly as before!

### CODE REVIEW

File: src/cascadia/TerminalSettingsModel/AllShortcutActions.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+// Include this file to be able to quickly define some code in the exact same
+// way for _every single shortcut action_. To use:
+//
+// 1. Include this file
+// 2. Define the ON_ALL_ACTIONS macro with what you want each action to show up
+//    as. Ex:
+//
+//    #define ON_ALL_ACTIONS(action) void action##Handler();
+//
+// 3. Then, use the ALL_SHORTCUT_ACTIONS macro to get the ON_ALL_ACTIONS marcro
```
Comment (zadjii-msft) [inline src/cascadia/TerminalSettingsModel/AllShortcutActions.h:18]:
```suggestion
// 3. Then, use the ALL_SHORTCUT_ACTIONS macro to get the ON_ALL_ACTIONS macro
```

**Review by DHowett — APPROVED** [review]

**Review by skyline75489 — APPROVED** [review]
Programming 101: less repeating, the better.

**Comment by zadjii-msft** [comment]
@miniksa @DHowett when did this get busted in `main`?

```
 67>C:\a\1\s\src\host\exe\exemain.cpp(70,22): error C2220: the following warning is treated as an error [C:\a\1\s\src\host\exe\Host.EXE.vcxproj]
    63>ClCompile:
         CopyToCharPopupTests.cpp
##[warning]src\host\exe\exemain.cpp(70,22): Warning C4456: declaration of 'dwValue' hides previous local declaration
         DbcsTests.cpp
    67>C:\a\1\s\src\host\exe\exemain.cpp(70,22): warning C4456: declaration of 'dwValue' hides previous local declaration [C:\a\1\s\src\host\exe\Host.EXE.vcxproj]
       C:\a\1\s\src\host\exe\exemain.cpp(60,11): message : see declaration of 'dwValue' [C:\a\1\s\src\host\exe\Host.EXE.vcxproj]

```

**Comment by miniksa** [comment]
> @miniksa @DHowett when did this get busted in `main`?
> 
> ```
>  67>C:\a\1\s\src\host\exe\exemain.cpp(70,22): error C2220: the following warning is treated as an error [C:\a\1\s\src\host\exe\Host.EXE.vcxproj]
>     63>ClCompile:
>          CopyToCharPopupTests.cpp
> ##[warning]src\host\exe\exemain.cpp(70,22): Warning C4456: declaration of 'dwValue' hides previous local declaration
>          DbcsTests.cpp
>     67>C:\a\1\s\src\host\exe\exemain.cpp(70,22): warning C4456: declaration of 'dwValue' hides previous local declaration [C:\a\1\s\src\host\exe\Host.EXE.vcxproj]
>        C:\a\1\s\src\host\exe\exemain.cpp(60,11): message : see declaration of 'dwValue' [C:\a\1\s\src\host\exe\Host.EXE.vcxproj]
> ```
> 
> EDIT: looks like it was fixed with a merge from inbox.

Yeah there was a kerfluffle where someone not in our team stealth changed console without telling us and this is just the fallout of discovering that and reconciling it. Should be good going forward.

**Review by miniksa — APPROVED** [review]
This is lovely. Let's merge it.

**Pull request merged**